test/test_StateMachineDevice.py

```diff
@@ -35,6 +35,17 @@ class MockStateMachineDevice(StateMachineDevice):
 
 
 class TestStateMachineDevice(unittest.TestCase):
+    def test_not_implemented_errors(self):
+        self.assertRaises(NotImplementedError, StateMachineDevice)
+
+        with patch('lewis.devices.StateMachineDevice._get_state_handlers',
+                   return_value={'test': Mock()}):
+            self.assertRaises(NotImplementedError, StateMachineDevice)
+
+            with patch('lewis.devices.StateMachineDevice._get_initial_state',
+                       return_value='test'):
+                self.assertRaises(NotImplementedError, StateMachineDevice)
```

Comment:
The first case you describe is the one I had in mind, this is why the methods are patched (so that they don't raise the exception before the other ones have a chance to do that) - I think that does what I wanted it to do.

You're right about the second case, this might happen if one of the bases of `StateMachineDevice` (`DeviceBase` and `CanProcessComposite`) decide to introduce methods that need to be implemented. A change like that basically changes the interface of the class, so the unit tests would likely require change as well.

Is there a better way to test this?